pkgs/applications/virtualization/singularity/generic.nix

```
@@ -0,0 +1,199 @@
+, version
+, src
+, projectName # "apptainer" or "singularity"
+  # These cannot be correctly override with `overrideAttrs`
+, vendorSha256 ? null
+, deleteVendor ? false
+, proxyVendor ? false
+, doCheck ? true
+, enableNvidiaContainerCli ? true
+  # This is required to run containers without suid
+, withoutSuid ? true
+, starterSuidPath ? null
+  # Remove the symlink to `singularity` when projectName != "singularity"
+, removeCompat ? true
+, extraDescription ? ""
+, extraMeta ? { }
+}:
+
+buildGoModule {
+  inherit pname version src;
+
+  # All three projects provide vendored source tarball
+  # so we set vendorSha256 = null;
+  # One who use unvendored source can override this
+  # with `override`
+  inherit vendorSha256 deleteVendor proxyVendor;
+
+  # apptainer and singularity-ce both want to reference Go
+  # to compile the extensions when building container images
+  allowGoReference = true;
+
+  passthru = {
+    inherit
+      projectName
+      removeCompat
+      withoutSuid
+      ;
+  };
+
+  buildInputs = [
+    gpgme
+    libuuid
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    cryptsetup
+    makeWrapper
+    util-linux
+    which
+  ];
+
+  postPatch = ''
+    substituteInPlace internal/pkg/build/files/copy.go \
+      --replace /bin/cp ${coreutils}/bin/cp
+  '';
+
+  configureScript = "./mconfig";
+
+  configureFlags = [
+    "--localstatedir=/var"
+  ]
+  ++ lib.optional withoutSuid "--without-suid";
+
+  preConfigure = ''
+    if ! [ -e .git ] || ! [ -e VERSION ]; then
+      echo "${version}" > VERSION
+    fi
+  '';
+
+  postConfigure = ''
+    patchShebangs .
+
+    # Patching the hard-coded [Dd]efaultPath
+    for subPath in cmd/internal/cli/actions.go internal/pkg/util/env/env.go e2e/env/env.go; do
+      if [ -f "$subPath" ]; then
+        echo "Patching [Dd]efaultPath in \"$subPath\""
+        sed -i 's|^\([ \t]*[Dd]efaultPath[ \t]*[:]\?=[ \t]*\)"[^"]*"|\1"${lib.makeBinPath [ coreutils squashfsTools bash ]}"|' "$subPath"
+      fi
```

Silently ignoring files which are not present could get quite hard to debug if things change in the future.

---

Replaced with `substituteInPlace`.